### DISCUSSION TITLE
<general coding bug not specific to regex>

### DISCUSSION THREAD

**Discussion by hanusek** [rust-lang/regex#796] [Q&A] [answered]
Hello.

I would like it to be taken from the text:
/dev/ttyS1
/dev/COM1
./ttyV1

like here:
[https://regexr.com/61uto
](https://regexr.com/61uto)

**Playground:**
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=ca8e65ebb7879558096130be8e6b7314

Please help me!

**Comment by BurntSushi** [comment, marked as answer]
Sorry, but I don't know what you mean by "not working." Are you getting an error message? If so, what does it say?

**Reply by hanusek** [reply]
It is not working correctly with for example: "/dev/ttyS1".

**Reply by BurntSushi** [reply]
Please provide a complete Rust program that I can compile and run. Please show me the output you're getting. Please show me the output you want.

If it helps, please avoid saying the words "not working" in your response.

**Reply by hanusek** [reply]
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=ca8e65ebb7879558096130be8e6b7314

**Reply by BurntSushi** [reply]
You're only checking the first two capturing groups. You need an `r3`.

**Reply by hanusek** [reply]
Thx.

**Comment by hanusek** [comment]
Solved.
